src/stream/src/executor/hash_agg.rs

```diff
@@ -83,6 +84,7 @@ struct HashAggExecutorExtra<S: StateStore> {
     key_indices: Vec<usize>,
 
     state_tables: Vec<RowBasedStateTable<S>>,
+    state_table_col_mappings: Vec<Vec<usize>>,
```

Comment:
could it be `Vec<(Vec<usize>, RowBasedStateTable<S>)>` ? And I think the mapping might be `Vec<Vec<Option<usize>>>` because not every column corresponds an input column. PTAL @BowenXiao1999 .

Reply:
Hmm, either LGTM. Indeed an Option is semantically better. So that we can assert col_mapping.len == table_columns.len.

Good for me to put it in another PR